### PR TITLE
Delaying Draft Pull Request Notifications until Marked as 'Ready for Review'

### PR DESCRIPTION
**Pull request by abdulsmapara**
#### Summary
Delaying Draft Pull Request Notifications until Marked as 'Ready for Review'

##### Requirement addressed:
- When a draft PR is submitted, an on-line notification gets posted.
- When a draft PR is marked as ready-for-review, a "standard" PR notification with the full body gets posted.

##### Description
When draft PR created, following message is posted, without PR description/body =>
image

When PR marked ready for review, following message is posted, which includes PR description/body =>
image

When PR directly created with ready for review, following message is posted, which includes PR description/body =>
image

#### Ticket Link
#680

### CODE REVIEW

File: server/plugin/template.go [hunk 1]
```diff
@@ -183,13 +183,39 @@ Assignees: {{range $i, $el := .Assignees -}} {{- if $i}}, {{end}}{{template "use
 {{- end -}}
 `))
 
-	template.Must(masterTemplate.New("newPR").Funcs(funcMap).Parse(`
+	template.Must(masterTemplate.New("newDraftPR").Funcs(funcMap).Parse(`
 {{ if eq .Config.Style "collapsed" -}}
 {{template "repo" .Event.GetRepo}} New pull request {{template "pullRequest" .Event.GetPullRequest}} was opened by {{template "user" .Event.GetSender}}.
```
Comment (hanzei) [inline server/plugin/template.go:188]:
```suggestion
{{template "repo" .Event.GetRepo}} New draft pull request {{template "pullRequest" .Event.GetPullRequest}} was opened by {{template "user" .Event.GetSender}}.
```

File: server/plugin/template.go [hunk 2]
```diff
@@ -183,13 +183,39 @@ Assignees: {{range $i, $el := .Assignees -}} {{- if $i}}, {{end}}{{template "use
...
 {{- else -}}
 #### {{.Event.GetPullRequest.GetTitle}}
 ##### {{template "eventRepoPullRequest" .Event}}
 #new-pull-request by {{template "user" .Event.GetSender}}
+{{- end }}
+`))
+
+	template.Must(masterTemplate.New("newReadyToReviewPR").Funcs(funcMap).Parse(`
```
Comment (hanzei) [inline server/plugin/template.go:196]: Let's keep the template name `newPR`. That also makes the diff easier to read.

File: server/plugin/template.go [hunk 3]
```diff
@@ -183,13 +183,39 @@ Assignees: {{range $i, $el := .Assignees -}} {{- if $i}}, {{end}}{{template "use
...
 `))
 
-	template.Must(masterTemplate.New("newPR").Funcs(funcMap).Parse(`
+	template.Must(masterTemplate.New("newDraftPR").Funcs(funcMap).Parse(`
 {{ if eq .Config.Style "collapsed" -}}
 {{template "repo" .Event.GetRepo}} New pull request {{template "pullRequest" .Event.GetPullRequest}} was opened by {{template "user" .Event.GetSender}}.
 {{- else -}}
 #### {{.Event.GetPullRequest.GetTitle}}
 ##### {{template "eventRepoPullRequest" .Event}}
 #new-pull-request by {{template "user" .Event.GetSender}}
```
Comment (hanzei) [inline server/plugin/template.go:192]: Per https://github.com/mattermost/mattermost-plugin-github/issues/680#issuecomment-1576257866 this should always render a one-line notification, i.e. the collapsed style should be used.

**Review by hanzei — CHANGES_REQUESTED** [review]
Thanks for the contribution :tada:

File: server/plugin/template.go (hunk 2 above)
Comment (abdulsmapara) [reply server/plugin/template.go]: for PR marked ready to review from draft, I have kept the name `markedReadyToReviewPR`, hope thats fine

File: server/plugin/webhook.go [hunk 4]
```diff
@@ -351,11 +352,11 @@ func (p *Plugin) postPullRequestEvent(event *github.PullRequestEvent) {
 	}
 
 	action := event.GetAction()
-	if action != actionOpened && action != actionLabeled && action != actionClosed {
+	if action != actionOpened && action != actionMarkedReadyForReview && action != actionLabeled && action != actionClosed {
 		return
 	}
-
```
Comment (hanzei) [inline server/plugin/webhook.go:357]: Nit: unrelated change

**Review by hanzei — APPROVED** [review]
Nice work :+1:

**Review by crspeller — APPROVED** [review]

**Comment by abdulsmapara** [comment]
@DHaussermann, requesting you to please review the PR.
cc: @hanzei

**Comment by abdulsmapara** [comment]
@DHaussermann, any updates on the PR review ?
cc: @hanzei, @crspeller

**Comment by DHaussermann** [comment]
Sorry @abdulsmapara for the delay. I will follow-up to find a testing resource for this PR.

**Comment by abdulsmapara** [comment]
no problem @DHaussermann 
Thanks for the update.

**Comment by abdulsmapara** [comment]
Any update @DHaussermann ?

cc: @hanzei ,@crspeller

**Comment by crspeller** [comment]
@lindalumitchell Do you have some time to review this PR?

**Comment by lindalumitchell** [comment]
@crspeller yes, I should be able to; I know Dylan is out next week as well. I'm unsure how to build a test version of the plugin though; can you help?

**Comment by hanzei** [comment]
@lindalumitchell You can download the bundle bundle at https://github.com/mattermost/mattermost-plugin-github/actions/runs/5855120323?pr=682.

**Review by lindalumitchell — APPROVED** [review]
Apologies for the further delay on this review; I hadn't tested this area before. But I got it all set up and tested and it looks great! Thanks very much for the contribution!

Verified as described that the simple notification posts on draft PR creation, and full notification posts when marked Ready for Review. Also smoke tested around other actions and notifications with all behaving as expected and no issues found. 👍

**Comment by hanzei** [comment]
Thanks @abdulsmapara :rocket:

**Comment by ThiefMaster** [comment]
Could this please be made configurable? 

I'd much rather have it show the full message when the PR is opened, and nothing at all when it's changed from/to draft (like it was before this PR).

In particular, it's spammy right now when someone opens the PR, then converts it to draft (because they forgot), and then back to a regular one shortly after (that results in a total of two "full" messages and one small).

I think this change would be more bearable if the plugin kept track of whether a PR was in "ready" state before so the bull message is only sent once. But even then I'd rather have a setting to disable the whole functionality, because depending on the project, Draft PR just means that there's still some stuff left for discussion, but that people still want to be properly informed about the PR...
